### PR TITLE
Update tests to match canton changes

### PR DESCRIPTION
**Pull request by samuel-williams-da**
Following PR on canton https://github.com/DACH-NY/canton/pull/13719, test tool needs to be updated for canton CI to pass.
This PR contains those changes, and sets canton to be that branch

Changes in `ledger/error/src/main/scala/com/daml/error` mirror thatt of the changes to canton, so please leave notes about that on the canton PR

### CODE REVIEW

File: ledger/error/src/main/scala/com/daml/error/ErrorResource.scala
Comment (samuel-williams-da) [inline ledger/error/src/main/scala/com/daml/error/ErrorResource.scala:1]: Do not review, see PR description

File: ledger/ledger-api-errors/src/main/scala/com/daml/error/definitions/groups/CommandExecution.scala
Comment (samuel-williams-da) [inline ledger/ledger-api-errors/src/main/scala/com/daml/error/definitions/groups/CommandExecution.scala:1]: Do not review, see PR description

File: ledger/ledger-api-errors/src/main/scala/com/daml/error/definitions/groups/ConsistencyErrors.scala
Comment (samuel-williams-da) [inline ledger/ledger-api-errors/src/main/scala/com/daml/error/definitions/groups/ConsistencyErrors.scala:1]: Do not review, see PR description

File: canton_dep.bzl [hunk 1]
```diff
@@ -4,5 +4,5 @@
 canton = {
     "sha": "a419de7d2e50578edf4c15bd4ed8f57d49396fa9832a02e9b78ff7a409d54326",
     "url": "https://www.canton.io/releases/canton-open-source-2.7.0-snapshot.20230705.10732.0.v8e5bbd5f.tar.gz",
-    "local": False,
+    "local": True,
```
Comment (atriantafyllos-da) [inline canton_dep.bzl:7]: why is this needed?

File: canton/BUILD.bazel [hunk 2]
```diff
@@ -55,17 +55,10 @@ java_import(
...
-    srcs = [":lib/canton.jar"] if canton["local"] else ["@canton//:jar"],
     outs = ["canton-lib-ee.jar"],
     cmd = """
 set -euo pipefail
 
-if [ "{local}" = "true" ]; then
-    cp $(location {src}) $@
-    exit 0
-fi
-
```
Comment (remyhaemmerle-da) [inline canton/BUILD.bazel:68]: @garyverhaegen-da There is an issue here. We cannot use the arbitrary sha for `canton-ee`.
I think using always the snapshot for EE is good enough.

Comment (samuel-williams-da) [reply canton/BUILD.bazel:68]: @remyhaemmerle-da Windows is now failing with `cp: target 'bazel-out/x64_windows-opt/bin/canton/canton-lib-ee.jar' is not a directory` - are these changes also needed in a ps1 file?

Comment (garyverhaegen-da) [reply canton/BUILD.bazel:68]: @remyhaemmerle-da I'm not sure I understand what you're trying to say here. Can you elaborate? What is preventing you fromusing the arbitrary sha for `canton-ee`?

Comment (samuel-williams-da) [reply canton/BUILD.bazel:68]: See these errors
https://dev.azure.com/digitalasset/daml/_build/results?buildId=145218&view=logs&j=52c0dd40-3465-5e8e-2f7c-ae591b3bddfa&t=199b63e7-8a1c-52a3-a4d5-f2bcbcf31b17&l=249

**Comment by samuel-williams-da** [comment]
@digital-asset/daml-docs Please review only the changes to [this file](docs/resources/generated-error-pages/error-codes-inventory.rst.inc). Note the changes that trigger this are in a different PR, so I cannot mark this as ready to review until that is merged.

File: docs/resources/generated-error-pages/error-codes-inventory.rst.inc [hunk 3]
```diff
@@ -758,6 +838,134 @@ DAML_INTERPRETER_INVALID_ARGUMENT
...
+
+
+
+
+.. _error_code_NON_COMPARABLE_VALUES:
+
+NON_COMPARABLE_VALUES
+---------------------------------------------------------------------------------------------------------------------------------------
+    
+    **Explanation**: This error occurs when you attempt to compare two values of different types using the builtin comparison types.
```
Comment (loiskilberg-da) [inline docs/resources/generated-error-pages/error-codes-inventory.rst.inc:895]:
```suggestion
    **Explanation**: This error occurs when you attempt to compare two values of different types using the built-in comparison types.
```

**Review by loiskilberg-da — APPROVED** [review]
just a few tiny suggestions. Looks good!

File: canton_dep.bzl (hunk 1 above)
Comment (samuel-williams-da) [reply canton_dep.bzl:7]: @remyhaemmerle-da Do you know if this is needed as well as the arbitrary_canton_sha file?

Comment (garyverhaegen-da) [reply canton_dep.bzl:7]: Yes, see README.

Comment (garyverhaegen-da) [reply canton_dep.bzl:7]: I may have misunderstood, so let me clarify. Setting `local` to `True` is necessary for CI to actually use the "local" Canton that gets built when `arbitrary_canton_sha` exists (and is a valid Canton commit sha). If you have the file but local is false, it gets built by CI but is then ignored by Bazel.

I do not have any opinion on whether or not you need an arbitrary commit here or you could use a published snapshot.

Comment (samuel-williams-da) [reply canton_dep.bzl:7]: I cannot use a published snapshot until tomorrow, as I merged the canton changes today. However if I wait until tomorrow, the canton bump here will fail. Instead I'm opting to use a custom hash until the bump tomorrow, which I believe will delete the sha file and revert local to false. Am I correct in that assumption?

Comment (garyverhaegen-da) [reply canton_dep.bzl:7]: You'll be the first to test it AFAIK, so who knows 🤷, but, yes, your assumption does described the intended and expected behaviour.

**Review by garyverhaegen-da — APPROVED** [review]
